src/contextplugin.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module core/contextplugin
+ */
+
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+import mix from '@ckeditor/ckeditor5-utils/src/mix';
+
+/**
+ * The base class for {@link module:core/context~Context} plugin classes.
```

Comment:
> A context plugin can either be initialized for an editor or for a context. In other words, it can either work within one editor instance or with one or more editor instances that use a single context. It's the context plugin's role to implement handling for both modes.

Is the above correct? Cause I'm not sure how it's supposed to work. If I'm correct, please add it to the docs. If not, explain me how it's supposed to work :D

Reply:
It's correct 👍

Reply:
I also thought that it'd be good to document rules that a context plugin must adhere to. I remember @pjasiun wrote down a couple of them. It's a good place to cover that.